distributed/comm/ucx.py

```diff
@@ -39,6 +43,50 @@ def init_once():
     import ucp as _ucp
 
     ucp = _ucp
+
+    # configuration of UCX can happen in two ways:
+    # 1) high level on/off flags which correspond to UCX configuration
+    # 2) explicity defined UCX configuration flags
+
+    # if any of the high level flags are set, as long as they are not Null/None,
+    # we assume we should configure basic TLS settings for UCX
+
+    if any(
+        [
+            dask.config.get("ucx.nvlink"),
+            dask.config.get("ucx.infiniband"),
+            dask.config.get("ucx.tcp-over-ucx"),
+            dask.config.get("ucx.net-devices"),
+        ]
+    ):
+        tls = "tcp,sockcm,cuda_copy"
+        tls_priority = "sockcm"
+
+        if dask.config.get("ucx.infiniband"):
+            tls = "rc," + tls
+        if dask.config.get("ucx.nvlink"):
+            tls = tls + ",cuda_ipc"
+
+        options = {"TLS": tls, "SOCKADDR_TLS_PRIORITY": tls_priority}
+
+        net_devices = dask.config.get("ucx.net-devices")
+        if net_devices is not None and net_devices != "":
+            options["NET_DEVICES"] = net_devices
+
+        dask.config.set({"ucx": options})
+
+    # support other clients (dask-cuda) passing ucx configurations directly
+    # when using explicty UCX configurations assume advanced users
+    elif all(
+        [
+            dask.config.get("ucx.TLS", False),
+            dask.config.get("ucx.SOCKADDR_TLS_PRIORITY", False),
+        ]
+    ):
+        pass
+    else:
+        dask.config.set({"ucx": {}})
```

Comment:
This is perhaps something to think about later, but it seems odd to delete user configuration.  

Instead, I wonder if a function like this might produce some new dictionary on demand, and that dictionary gets used in the `ucp.init` call without modifying the dask.config global in-place.

Reply:
I went back and forth on this as well.  I think a new dictionary makes sense.  i'd like to do that now